specifications_dbus.md

```diff
@@ -0,0 +1,92 @@
+# Specifications
+
+## Index
+
+* [Connection](#connection)
+* [Connector](#connector)
+* [Connector API](#connector-API)
+* [Distributor](#distributor)
+* [Distributor API](#distributor-api)
+* [Appendix: Implementation practices](#appendix-implementation-practices)
+
+## Connection
+
+A *connection* is created when a connector registers with a distributor. It is uniquely identified by its *token*, a random string which the connector generates as part of the registration process.
+
+## Connector
+
+A *connector* is an application, or software library used in an application, which wants to receive push messages from a UnifiedPush distributor.
+
+When a connector first wants to register, it should fetch all service names currently registered on the session bus, and filter it to only those which begin with `org.unifiedpush.Distributor.`. It should then choose one of these (see [Appendix: Implementation practices](#appendix-implementation-practices)) to use as its distributor. It must then generate a random string to use as a token, and call the [org.unifiedpush.Distributor1.Register](#orgunifiedpushdistributor1register-string--string-string) method with this token as the argument. If the return value indicates "NEW_ENDPOINT", the connector must then save the token and the endpoint received in its persistent storage. Registration is now complete.
+
+On subsequent startups, the connector should call [org.unifiedpush.Distributor1.Register](#orgunifiedpushdistributor1register-string--string-string) with the *same* token, and store the new endpoint returned, if it has changed. The connector should also implement the [org.unifiedpush.Connector1.NewEndpoint](#orgunifiedpushconnector1newendpoint-string-string--nothing) method to receive notification about a changed endpoint from the distributor at any time.
+
+The application SHOULD provide an activatable D-Bus service, so that it can be activated by a new push message when it is not running. See [Appendix: Implementation practices](#appendix-implementation-practices).
+
+## Connector API
+
+Connectors MUST provide a service with their application ID, that is, the same name as the name that they sent their `org.unifiedpush.Distributor1.Register` call from.
```

Comment:
This is not very clear. It should be defined on `org.unifiedpush.Distributor1.Register` side